polling_stations/apps/data_importers/data_types.py

```diff
@@ -323,7 +324,8 @@ def remove_records_that_dont_match_addressbase(self, addressbase_data):
             ):
                 continue
             else:
-                self.elements.remove(input_record)
+                to_remove.append(input_record)
+        self.elements = [e for e in self.elements if e not in to_remove]
```

Comment:
Not sure if this matters, but you could use sets for `to_remove` and `self.elements` and then do 

```python
self.elements = self.elements - to_remove
```

If there's a reason for `self.elements` to be a list, then never mind :)

Reply:
I think there's an `AddressList` used to be an `AddressSet` -> https://github.com/DemocracyClub/UK-Polling-Stations/commit/4949ddaaa69b1d416da82ba930398c6279404d1e
TBH I just stuck with the framework that was there :). 
I'll make a note to revisit whether there'd be any advantage to going back to a set when there's less imports to write!